components/omega/test/CMakeLists.txt

```diff
@@ -160,6 +160,42 @@ add_test(
   COMMAND ${MPI_EXEC} -n 8 -- ./${_TestDecompName}
 )
 
+#############
+# HorzMesh test
+#############
+
+set(_TestHorzMeshName testHorzMesh.exe)
+
+# Add broadcast test
+add_executable(${_TestHorzMeshName} ocn/HorzMeshTest.cpp)
+
+target_include_directories(
+  ${_TestHorzMeshName}
+  PRIVATE
+  ${OMEGA_SOURCE_DIR}/src/base
+  ${OMEGA_SOURCE_DIR}/src/infra
+  ${OMEGA_SOURCE_DIR}/src/ocn
+  ${PARMETIS_ROOT}/include
+)
+
+target_compile_options(
+  ${_TestHorzMeshName}
+  PRIVATE
+  ${OMEGA_CXX_FLAGS}
+)
+
+target_link_options(
+  ${_TestHorzMeshName}
+  PRIVATE
+  ${OMEGA_LINK_OPTIONS}
+)
+
+target_link_libraries(${_TestHorzMeshName} ${OMEGA_LIB_NAME} yakl metis)
+
+add_test(
+  NAME HORZMESH_TEST
+  COMMAND ${MPI_EXEC} -n 8 -- ./${_TestHorzMeshName}
+)
```

Comment:
You will also need to add HORZMESH_TEST to the set_test_properties list at the end of this file